pkg/ingester/ingester.go

```diff
@@ -1295,48 +1314,69 @@ func (i *Ingester) pushSamplesToAppender(userID string, timeseries []mimirpb.Pre
 		// we actually ingested all samples which haven't failed.
 		switch {
 		case errors.Is(err, storage.ErrOutOfBounds):
+			if cat != nil {
+				cat.IncrementDiscardedSamples(mimirpb.FromLabelAdaptersToLabels(labels), 1, reasonSampleOutOfBounds, startAppend)
+			}
```

Comment:
Instead of repeating the check everywhere, we can make a nil `costattribution.Tracker` be a noop.

Reply:
addressed in commit https://github.com/grafana/mimir/pull/9733/commits/f4ebc07871ef79a0ed7ee31c48032a9fd42fe757#diff-8e51742acf8f88fa2ae01855fb19125feb5c54c41023f1eaa8beded9f1b4bcc3R1115